src/frontend/Typechecker.ml

```diff
@@ -1071,13 +1071,11 @@ let verify_sampling_distribution loc tenv id arguments =
     SignatureMismatch.matching_function tenv (name ^ suffix) argumenttypes in
   let sampling_dists =
     List.map ~f:name_w_suffix_sampling_dist Utils.distribution_suffices in
-  let is_sampling_dist_defined =
+  if
     List.exists
       ~f:(function UniqueMatch (ReturnType UReal, _, _) -> true | _ -> false)
       sampling_dists
-    && name <> "binomial_coefficient"
-    && name <> "multiply" in
-  if is_sampling_dist_defined then ()
+  then ()
   else
     match
```

Comment:
You're right, this could be much simpler. I think the reason we needed to search before is that there could potentially be a `_log` and an `_lpdf`, one of which was correct and one of which wasn't? Or perhaps it was always over-complicated.

The return type is always going to be correct because of the `verify_fundef_dist_rt` check which occurs earlier